src/DataSource.ts

```diff
@@ -21,9 +28,22 @@ export class DataSource extends DataSourceApi<CmkQuery> {
   }
 
   async query(dataQueryRequest: DataQueryRequest<CmkQuery>): Promise<DataQueryResponse> {
+    for (const target of dataQueryRequest.targets) {
+      target.requestSpec = replaceVariables(target.requestSpec, dataQueryRequest.scopedVars);
+    }
     return this.getBackend().query(dataQueryRequest);
   }
 
+  async metricFindQuery(query: MetricFindQuery, options?: any): Promise<MetricFindValue[]> {
+    if (query.objectType === 'site') {
+      // rest-api site endpoint were added in 2.2.0 so we have to use the web-api here
+      // TODO: clean up (remove filterSites from Backend) with end of 2.1.0
+      return await this.getBackend().filterSites();
+    }
+    // we use the rest backend for both web and rest backend, because those endpoints are already implement in 2.1.0
+    return await this.restBackend.metricFindQuery(query);
```

Comment:
In the future, we should think about moving the `filter*` functions to a class outside the backends. The Datasource shouldn't be the one to decide where to get its data from.

Reply:
added to the todo list